devlib/utils/ssh.py

```diff
@@ -299,6 +312,469 @@ def __init__(self,
+                 password=None,
+                 port=None,
+                 timeout=None,
+                 password_prompt=None,
+                 original_prompt=None,
+                 ):
+        # TODO: need to insert check that host is the actual place
+        # where devlib is now running
+        self.host = host
+        if username is not None or username is not 'root':
+            gem5_logger.info("User is always root in gem5")
+        if password is not None or password is not '':
+            gem5_logger.info("No password needed in gem5")
+        self.username =  'root'
+        self.password = None
+        self.port = -1
+        # Long timeouts to account for gem5 being slow
+        self.default_timeout = 3600
+        self.ready_timeout = self.default_timeout * 3
+        # Intermediate directory to push things to gem5 using VirtIO
+        self.gem5interactdir = None
+        # Directory to store things from gem5 on the host
+        self.gem5outdir = None
+        self.busybox_wrapper = None
+        self.m5_path = None
+        # Actual telnet connection to gem5 simulation
+        self.conn = None
+        # Flag to indicate the gem5 device is ready to interact with the
+        # outer world
+        self.ready = False
+
+        # Connect to gem5
+        self._connect_gem5(platform)
+
+        # Wait for boot
+        self._wait_for_boot()
+
+        # Mount the virtIO to transfer files in/out gem5 system
+        self._mount_virtio()
+
+        # Ready for interaction now
+        self.ready = True
+
+
+    def set_busybox_wrapper(self, wrapper):
+        self.busybox_wrapper = wrapper
+
+    def connected_as_root(self):
+        # gem5 is always rooted
+        return True
+
+    def push(self, source, dest, timeout=None, wrap_busybox=False):
+        """
+        Push a file to the gem5 device using VirtIO
+
+        The file to push to the device is copied to the temporary directory on
+        the host, before being copied within the simulation to the destination.
+        Checks, in the form of 'ls' with error code checking, are performed to
+        ensure that the file is copied to the destination.
+        """
+        filename = os.path.basename(source)
+        logger.debug("Pushing {} to device.".format(source))
+        logger.debug("gem5interactdir: {}".format(self.gem5interactdir))
+        logger.debug("dest: {}".format(dest))
+        logger.debug("filename: {}".format(filename))
+
+        # We need to copy the file to copy to the temporary directory
+        self._move_to_temp_dir(source)
+
+        # Back to the gem5 world
+        self._gem5_shell("ls -al /mnt/obb/{}".format(filename))
+        if wrap_busybox:
+            self._gem5_shell("{} cp /mnt/obb/{} {}".format(self.busybox_wrapper, filename, dest))
+        else:
+            self._gem5_shell("cat /mnt/obb/{} > {}".format(filename, dest))
+        self._gem5_shell("sync")
+        self._gem5_shell("ls -al {}".format(dest))
```

Comment:
OK, so whilst this is likely OK to use, I think that we should change it. Under Linux, it would be fine, but for Android `/mnt/obb` is used as a temporary file system in certain situations. I think that we should change this to something else if possible. How about something along the lines of `/mnt/host`, but we would need to make sure that we are su at the point where we create and mount this node. Any thoughts on the name?

For the record, we would need to make the same change in WA.